fundamentals/middleware/rate-limit/WebRateLimitAuth/SampleRateLimiterPolicy.cs

```diff
@@ -0,0 +1,33 @@
+using System.Threading.RateLimiting;
+using Microsoft.AspNetCore.RateLimiting;
+
+namespace WebRateLimitAuth;
+
+public class SampleRateLimiterPolicy : IRateLimiterPolicy<string>
+{
+    private Func<OnRejectedContext, CancellationToken, ValueTask>? _onRejected;
+
+    public SampleRateLimiterPolicy(ILogger<SampleRateLimiterPolicy> logger)
+    {
+        _onRejected = (context, token) =>
+        {
+            context.HttpContext.Response.StatusCode = 429;
+            logger.LogWarning($"Request rejected by {nameof(SampleRateLimiterPolicy)}");
+            return ValueTask.CompletedTask;
+        };
+    }
+
+    public Func<OnRejectedContext, CancellationToken, ValueTask>? 
+                                                         OnRejected { get => _onRejected; }
```

Comment:
Is it OK to have `OnRejected`  here and also in `RateLimiterOptions` directly?

Reply:
Right, policy OnRejected will win.